src/Microsoft.Health.Fhir.Api/Controllers/AadSmartOnFhirProxyController.cs

```diff
@@ -45,19 +42,19 @@ public class AadSmartOnFhirProxyController : Controller
         /// <summary>
         /// Initializes a new instance of the <see cref="AadSmartOnFhirProxyController" /> class.
         /// </summary>
-        /// <param name="securityConfiguration">Security configuration parameters.</param>
+        /// <param name="securityConfigurationOptions">Security configuration parameters.</param>
         /// <param name="httpClientFactory">HTTP Client Factory.</param>
         /// <param name="logger">The logger.</param>
-        public AadSmartOnFhirProxyController(IOptions<SecurityConfiguration> securityConfiguration, IHttpClientFactory httpClientFactory, ILogger<SecurityConfiguration> logger)
+        public AadSmartOnFhirProxyController(IOptions<SecurityConfiguration> securityConfigurationOptions, IHttpClientFactory httpClientFactory, ILogger<SecurityConfiguration> logger)
         {
-            EnsureArg.IsNotNull(securityConfiguration, nameof(securityConfiguration));
+            EnsureArg.IsNotNull(securityConfigurationOptions, nameof(securityConfigurationOptions));
 
-            _securityConfiguration = securityConfiguration.Value;
-            _isAadV2 = new Uri(_securityConfiguration.Authentication.Authority).Segments.Contains("v2.0");
-            _logger = logger;
+            SecurityConfiguration securityConfiguration = securityConfigurationOptions.Value;
+            _isAadV2 = new Uri(securityConfiguration.Authentication.Authority).Segments.Contains("v2.0");
             _httpClientFactory = httpClientFactory;
+            _logger = logger;
 
-            var openIdConfigurationUrl = $"{_securityConfiguration.Authentication.Authority}/.well-known/openid-configuration";
+            var openIdConfigurationUrl = $"{securityConfiguration.Authentication.Authority}/.well-known/openid-configuration";
 
             HttpResponseMessage openIdConfigurationResponse;
```

Comment:
There are multiple places where OpenId configuration is being retrieved and processed. We should centralize it into a separate class. Filed #385 to track this issue.